internal/terraform/cmd_drift.go

```diff
@@ -0,0 +1,380 @@
+	}
+
+	if verbose {
+		slog.SetLogLoggerLevel(slog.LevelDebug)
+	}
+
+	// if path isn't specified, we're assuming metronome-substrate
+	if path == "" {
+		var err error
+		path, err = findGitRoot()
+		if err != nil {
+			log.Fatalf("Failed to find git root: %s", err)
+		}
+		path = filepath.Join(path, "root-modules")
+		_, err = os.Stat(path)
+		if err != nil {
+			log.Fatalf("Run from metronome-substrate or specify a path: %s", err)
+		}
+	}
+
+	calculateDrift(path, matchPatterns, skipPatterns)
+}
+
+type config struct {
+	SkipRootModules []string `yaml:"skip_root_modules"`
+	Concurrency     int      `yaml:"concurrency"`
+}
+
+type planOutput struct {
+	Changed bool
+	Error   error
+	Output  io.ReadWriter
+	Path    string
+}
+
+type planInput struct {
+	Path string
+	Lock bool
+}
+
+func calculateDrift(path string, matchPatterns []string, skipPatterns []string) {
+	ctx := context.Background()
+
+	modules, err := getModules(path)
+	if err != nil {
+		log.Fatalf("Failed to get root modules: %s", err)
+	}
+
+	var matchedModules []string
+	for _, module := range modules {
+		for _, pattern := range matchPatterns {
+			match, err := doublestar.PathMatch(fmt.Sprintf("**/%s/**", pattern), module)
+			if err != nil {
+				log.Fatalf("match pattern error: %s", err)
+			}
+			if match {
+				matchedModules = append(matchedModules, module)
+			}
+		}
+	}
+	var finalModules []string
+	for _, module := range matchedModules {
+		for _, pattern := range skipPatterns {
+			match, err := doublestar.PathMatch(fmt.Sprintf("**/%s/**", pattern), module)
+			if err != nil {
+				log.Fatalf("skip pattern error: %s", err)
+			}
+			if !match {
+				finalModules = append(finalModules, module)
+			}
+		}
+	}
+
+	planCount := len(finalModules)
+	slog.Debug(
+		"SEARCH",
+		slog.Int("moduleCount", len(modules)),
+		slog.Any("matchPatterns", matchPatterns),
+		slog.Int("matchCount", len(matchedModules)),
+		slog.Any("skipPatterns", skipPatterns),
+		slog.Int("skipCount", len(matchedModules)-len(finalModules)),
+	)
+	slog.Info("running plans", "planCount", len(finalModules))
+
+	terraformExec, err = installTerraform(ctx)
+	if err != nil {
+		log.Fatalf("Failed to install terraform: %s", err)
+	}
+
+	inputChannel := make(chan planInput, planCount)
+	outputChannel := make(chan planOutput, planCount)
+	if DefaultConcurrency > planCount {
+		DefaultConcurrency = planCount
+	}
+	for i := 0; i < DefaultConcurrency; i++ {
+		go func(workerId int) {
+			planner(ctx, workerId, inputChannel, outputChannel)
+		}(i)
+	}
+
+	for _, rootModule := range finalModules {
+		slog.Debug("added to inputChannel", "module", rootModule)
+		slog.Debug("added to inputChannel", "module", cleanRootModulePath(rootModule))
+		inputChannel <- planInput{Path: rootModule, Lock: false}
+	}
+	close(inputChannel)
+
+	clean := 0
+	changes := 0
+	errors := 0
+	for i := 0; i < planCount; i++ {
+		result := <-outputChannel
+		switch {
+		case result.Error != nil:
+			errors++
+			color.Red(fmt.Sprint(result))
```

Comment:
Calling `String()` via `fmt.Sprint` just made me think there wasn't a good representation of the object. IMO better to just call it directly

```suggestion
			color.Red(result.String())
```